### PR TITLE
pull request review guidelines

### PR DESCRIPTION
**Pull request by jodygarnett**

### CODE REVIEW

**Comment by jodygarnett** [comment]
Feedback form meeting is to remove references to "patch" and make the procedures consistent :(

File: doc/en/developer/source/policies/psc.rst [hunk 1]
```diff
@@ -19,6 +19,64 @@ The PSC is made up of individuals who are intended to represent the various comm
...
+* Gabriel Roldan
+* Phil Scadden
+
+PSC Membership
+--------------
+
+New PSC members
+^^^^^^^^^^^^^^^
+
+A new PSC member can be nominated at any time.  Voting for a new PSC is done by current active PSC members.  There is no hard limit to the number of PSC members, but we want a relatively active PSC.  PSC nominations are generally given in recognition to very significant contributions to the project.  Membership is open to non-technical people, for example if someone is to make huge advances to the documentations or marketing of GeoServer, for example.
```
Comment (bradh) [inline doc/en/developer/source/policies/psc.rst]: documentations -> documentation

File: doc/en/developer/source/policies/pull_request.rst [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+Pull Requests
+=============
+
+As with most open source project GeoServer is very happy to accept pull requests from the community. Contributions from the community have been a source of some of the best bug fixes and improvements in GeoServer and is a great way to give back to the project.
+
+This document is a guide designed to help contributors in successfully submitting a pull request:
+
+.. note:: 
+
+   We ask all fixes applied to master first, and then back ported to the stable and maintenance branches.
```
Comment (bradh) [inline doc/en/developer/source/policies/pull_request.rst]: We ask *that* all fixes *be* applied

File: doc/en/developer/source/policies/pull_request.rst [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+.. note:: 
+
+   We ask all fixes applied to master first, and then back ported to the stable and maintenance branches. 
+   
+   We require the following for fixes submitted via pull requests:
+
+   1. Required: Prepare your fix for the master branch
+   2. Required: Check the following (and comment in your pull request, or jira issue):
+      
+      * Investigate if the issue impact all the active branches (master / stable / maintenance)
```
Comment (bradh) [inline doc/en/developer/source/policies/pull_request.rst]: all -> any of

File: doc/en/developer/source/policies/pull_request.rst [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+      * Investigate the your fix can be backported to all the active branches (master / stable / maintenance)
```
Comment (bradh) [inline doc/en/developer/source/policies/pull_request.rst]: * Investigate whether the fix can be backported to the applicable active branches (master / stable / maintenance)

File: doc/en/developer/source/policies/pull_request.rst [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+     git push origin my_bugfix
+
+#. Visit your github repo page and issue the pull request. 
+
+#. For the release notes we ask that you open a `JIRA <https://osgeo-org.atlassian.net/projects/GEOS>`_ ticket linking to your pull request.
+
+   At this point the core developers will be notified of the pull request and review it at the earliest 
+convenience.
+
+Core developers will :ref:`review` the patch and may equire changes or improvements to it prior to it being accepted. It will be up to the submitter to amend the pull request and keep it alive until it gets merged. Please be patient as pull requests are often reviewed in spare time so turn-around can be a little slow.
```
Comment (bradh) [inline doc/en/developer/source/policies/pull_request.rst]: equire -> require

File: doc/en/developer/source/policies/pull_request.rst [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+
+If a pull request becomes stale with no feedback from the submitter for a couple of months long, it will linked form a JIRA issue (to avoid losing the partial work) and then be closed.
```
Comment (bradh) [inline doc/en/developer/source/policies/pull_request.rst]: linked form -> linked to from

File: doc/en/developer/source/policies/pull_request.rst [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+
+In general developers will only accept pull requests that apply cleanly against the latest versions of the stable and master branches. Therefore before you generate a patch it is important that you update your checkout to the latest for the branch you are working on.
+
+Ensure your IDE/editor is properly configured
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Ensure that your development environment is properly configured for GeoServer development. A common issue 
+with patches from newcomers is that their text editor is configured to use tabs rather than spaces.
+
+See the the :ref:`eclipse_guide` for general information about formatting and ide setup.
```
Comment (bradh) [inline doc/en/developer/source/policies/pull_request.rst]: ide -> IDE

File: doc/en/developer/source/policies/pull_request.rst [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+Fix one thing at a time
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Do not batch up multiple unrelated changes into a single patch. If you want to fix multiple issues work
+on them separately and submit separate patches for them.
+
+Be patient
+^^^^^^^^^^
+
+The core developers review community patches in spare time. Be cognizant of this and realize that just
```
Comment (aaime) [inline doc/en/developer/source/policies/pull_request.rst]: Well, the contributor might be contributing in paid time too (happens a lot), it's the unlucky reviewer that gets  nothing out of it.

File: doc/en/developer/source/policies/review.rst [hunk 9]
```diff
@@ -0,0 +1,76 @@
+.. _review:
+
+Review
+======
+
+Reviewing code is an invaluable process that is used to ensure that the highest quality code possible gets committed into GeoServer at all times.
```
Comment (aaime) [inline doc/en/developer/source/policies/review.rst]: I'd go for "acceptable" quality code. Highest is far away from what we have and what we demand.

File: doc/en/developer/source/policies/review.rst [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+
+.. note:: Providing an initial check, while better than no follow up at all, may seem petty and discouraging to the contributor. It is a good idea to state that this is only quick feedback - while either more time or an appropriate expert is found.
+
+* *Contribution agreement* on file. Check if the contribution requires a contribution agreement, and if it matches the requisites, verify that a contribution agreement has been provided to OSGeo, if not, demand for one.
+
+* *Presence of tests*. Given a large code base, that pretty much nobody really can hope to master fully, the large number of external contributors, as well as focused changes to the code base, and the fast evolution of the code base, tests are really the only line of defense against accidental breakage of the contributed functionality. That is why we always demand to have at least one test, it's not a "punishment", but a basic guarantee that the fix or new functionality being contributed will still be there, and working, a bare few months. So always check there is at least one test, one that would break in a evident way if there ever was a regression in the contributed code.
```
Comment (aaime) [inline doc/en/developer/source/policies/review.rst]: "a bare few month" -> "a bare few months later"

File: doc/en/developer/source/policies/review.rst [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+
+* *Presence of documentation*. For functional, user or protocol facing changes, check that some basic documentation has been contributed. Almost will know the functionality is there if we don't have documentation on how it works. As a plus, documentation will help you understand quicker what is getting contributed, both from a high level view, and as a guide though the changes being made.
```
Comment (aaime) [inline doc/en/developer/source/policies/review.rst]: Almost will know -> Almost nobody will know

File: doc/en/developer/source/policies/review.rst [hunk 12]
```diff
@@ -0,0 +1,76 @@
...
+Presence of a proposal, if required. For any large change our process demands a formal proposal to be made and discussed in the community before a pull request gets made. If the pull request is made anyways, point them to the process and warn that the community might eventually request changes in the approach
```
Comment (aaime) [inline doc/en/developer/source/policies/review.rst]: *Presence of a proposal* (italic)

**Comment by aaime** [comment]
Added some more feedback, once fixed, I'd say this one is good to go

File: doc/en/developer/source/policies/pull_request.rst (hunk 8 above)
Comment (jodygarnett) [inline doc/en/developer/source/policies/pull_request.rst]: Here is revised wording to take that into account:

````
We understand professionals working on the code base may meet a customer or contractual obligation, please review `commercial support <http://geoserver.org/support/>`__ options if you are not in a position to wait for a volunteer.
```

**Pull request merged**